TKeazirian.HTTPServer.Tests/Router/RouterTests.cs

```diff
@@ -28,33 +32,51 @@ public void RouterRoutesToExpectedHandler()
     [InlineData("/")]
     public void ResourceNotFoundHandlerCalledWhenPathIsNotConfigured(string path)
     {
-        HTTPServer.Request.Request testRequest = new HTTPServer.Request.Request("GET", path, "", "");
-        var testRoutesConfig = new RoutesConfig(new Dictionary<string, HTTPServer.Handler.Handler>
+        Request testRequest = new Request("GET", path, "", "");
+        var testRoutesConfig = new RoutesConfig(new Dictionary<string, Handler>
         {
             { "/test_path", new MockHandler() }
         });
         Router router = new Router(testRoutesConfig);
 
-        HTTPServer.Response.Response response = router.Route(testRequest);
+        Response response = router.Route(testRequest);
 
         Assert.Equal("HTTP/1.1 404 Not Found\r\n", response.ResponseStatusLine);
     }
 
     [Fact]
     public void IfGetMethodIsAllowedOnEndpointThenHeadMethodIsAllowed()
     {
-        HTTPServer.Request.Request testRequest = new HTTPServer.Request.Request("HEAD", "/test_path", "", "");
+        Request testRequest = new Request("HEAD", "/test_path", "", "");
         string testHeaders = "Content-Type: text/plain\r\nContent-Length: 9\r\n\r\n";
-        var testRoutesConfig = new RoutesConfig(new Dictionary<string, HTTPServer.Handler.Handler>
+        var testRoutesConfig = new RoutesConfig(new Dictionary<string, Handler>
         {
             { "/test_path", new MockHandler() }
         });
         Router router = new Router(testRoutesConfig);
 
-        HTTPServer.Response.Response response = router.Route(testRequest);
+        Response response = router.Route(testRequest);
 
         Assert.Equal("HTTP/1.1 200 OK\r\n", response.ResponseStatusLine);
         Assert.Equal(testHeaders, response.ResponseHeaders);
         Assert.Empty(response.GetBody());
     }
+
+    [Fact]
+    public void IfImproperFormattedRequest404IsCalled()
+    {
+        Request badTestRequest = new Request("", "GET", HelperFunctions.CreateTestResponseHeaders(), "/test_path");
+        var testRoutesConfig = new RoutesConfig(new Dictionary<string, Handler>
+        {
+            { "/test_path", new MockHandler() },
+            { "/unhappy_path", new MockResourceNotFoundHandler() }
+        });
+
+        Router router = new Router(testRoutesConfig);
+
+        Response response = router.Route(badTestRequest);
+
+        Assert.Equal("HTTP/1.1 404 Not Found\r\n", response.ResponseStatusLine);
+        Assert.Equal("The resource cannot be found", response.ResponseBody);
```

Comment:
for this assertion, would asserting the status code be enough?  The rest appear to be implementation details.